avm-transpiler/src/transpile.rs

```diff
@@ -321,6 +304,84 @@ fn handle_foreign_call(
     });
 }
 
+fn set_for_uint(tag: AvmTypeTag, dest: &MemoryAddress, value: u128) -> AvmInstruction {
+    AvmInstruction {
+        opcode: AvmOpcode::SET,
+        indirect: Some(0),
+        tag: Some(tag),
+        operands: vec![
+            // const
+            match tag {
+                AvmTypeTag::UINT8 => AvmOperand::U8 { value: value as u8 },
+                AvmTypeTag::UINT16 => AvmOperand::U16 {
+                    value: value as u16,
+                },
+                AvmTypeTag::UINT32 => AvmOperand::U32 {
+                    value: value as u32,
+                },
+                AvmTypeTag::UINT64 => AvmOperand::U64 {
+                    value: value as u64,
+                },
+                AvmTypeTag::UINT128 => AvmOperand::U128 { value: value },
+                _ => panic!("Invalid type tag {:?} for set", tag),
+            },
+            // dest offset
+            AvmOperand::U32 {
+                value: dest.to_usize() as u32,
+            },
+        ],
+    }
+}
+
+fn cast(
+    source: &MemoryAddress,
+    destination: &MemoryAddress,
+    destination_tag: AvmTypeTag,
+) -> AvmInstruction {
+    AvmInstruction {
+        opcode: AvmOpcode::CAST,
+        indirect: Some(0),
+        tag: Some(destination_tag),
+        operands: vec![
+            AvmOperand::U32 {
+                value: source.to_usize() as u32,
+            },
+            AvmOperand::U32 {
+                value: destination.to_usize() as u32,
+            },
+        ],
+    }
+}
+
+fn handle_set(
+    avm_instrs: &mut Vec<AvmInstruction>,
+    destination: &MemoryAddress,
+    value: &Value,
+    bit_size: &u32,
+) {
+    let tag = tag_from_bit_size(bit_size);
+
+    if !matches!(tag, AvmTypeTag::FIELD) {
+        avm_instrs.push(set_for_uint(tag, destination, value.to_u128()));
+    } else {
+        // Handling fields is a bit more complex since we cannot fit a field in a single instruction.
+        // We need to split the field into 128-bit chunks and set them individually.
+        let field = value.to_field();
+        if !field.fits_in_u128() {
```

Comment:
Should we even care about `SET` for field type? I think that's still a "nice-to-have" feature, but probably don't care to support it for now?